src/components/Admin/Admin.js

```diff
@@ -16,45 +18,42 @@ class Admin extends Component {
   };
 
   authenticate = () => {
-    firebase
-      .auth()
-      .getRedirectResult()
-      .then(result => {
-        if (result.user) {
-          const userRef = firebase
-            .firestore()
-            .collection('users')
-            .doc(result.user.uid);
-
-          userRef.get().then(user => {
-            this.setState({loggedIn: true});
-            if (user.exists) {
-              if (user.data().admin) {
-                this.initAdmin(user.data().name);
-              } else {
-                userRef.set({
-                  name: result.user.displayName,
-                  admin: false
-                });
-                this.setState({status: 'You are not an administrator.'});
-              }
-            } else {
-              this.setState({status: 'You are not an administrator.'});
-            }
-          });
-        } else {
-          this.setState({status: 'You are not authenticated.'});
-        }
-      })
-      .catch(function(error) {
-        this.setState({status: error});
-      });
+    firebase.auth().onAuthStateChanged(this.loggedIn);
   };
 
   login = () => {
     firebase.auth().signInWithRedirect(new firebase.auth.GoogleAuthProvider());
   };
 
+  loggedIn = user => {
+    if (!user) {
+      this.setState({status: 'You are not authenticated'});
+      return;
+    }
+
+    const userRef = firebase
+      .firestore()
+      .collection('users')
+      .doc(user.uid);
+
+    userRef.get().then(u => {
```

Comment:
Is there a more descriptive name we could use here? I'm presuming `u` is used since `user` is already used above

What's the difference between them, can we suffix the two with what they acutally are? e.g. `userEntry`